src/framework/vst/internal/vstinstancesregister.cpp

```diff
@@ -64,44 +69,52 @@ IVstInstancePtr VstInstancesRegister::makeAndRegisterMasterFxPlugin(const muse::
 
 void VstInstancesRegister::registerInstrPlugin(const muse::audio::TrackId trackId, IVstInstancePtr instance)
 {
-    ONLY_AUDIO_THREAD(threadSecurer);
+    IF_ASSERT_FAILED(instance) {
+        return;
+    }
 
     std::lock_guard lock(m_mutex);
 
-    m_vstiPluginsMap.insert_or_assign(trackId, instance);
+    m_instances.insert_or_assign({ Type::Instrument, instance->resourceId(), trackId, 0 }, instance);
 }
 
-void VstInstancesRegister::registerFxPlugin(const muse::audio::TrackId trackId, const muse::audio::AudioResourceId& resourceId,
-                                            const AudioFxChainOrder chainOrder, IVstInstancePtr instance)
+void VstInstancesRegister::registerFxPlugin(const muse::audio::TrackId trackId,
+                                            const AudioFxChainOrder chainOrder,
+                                            IVstInstancePtr instance)
 {
-    ONLY_AUDIO_THREAD(threadSecurer);
+    IF_ASSERT_FAILED(instance) {
+        return;
+    }
 
     std::lock_guard lock(m_mutex);
 
-    VstFxInstancesMap& instancesMap = m_vstFxPluginsMap[trackId];
-    instancesMap.insert_or_assign(std::make_pair(resourceId, chainOrder), instance);
+    m_instances.insert_or_assign({ Type::Effect, instance->resourceId(), trackId, chainOrder }, instance);
 }
 
-void VstInstancesRegister::registerMasterFxPlugin(const muse::audio::AudioResourceId& resourceId, const AudioFxChainOrder chainOrder,
-                                                  IVstInstancePtr instance)
+void VstInstancesRegister::registerMasterFxPlugin(const AudioFxChainOrder chainOrder, IVstInstancePtr instance)
 {
-    ONLY_AUDIO_THREAD(threadSecurer);
+    registerFxPlugin(-1, chainOrder, instance);
+}
 
+IVstInstancePtr VstInstancesRegister::instanceById(const VstInstanceId id) const
+{
     std::lock_guard lock(m_mutex);
-
-    m_masterPluginsMap.insert_or_assign(std::make_pair(resourceId, chainOrder), instance);
+    for (const auto& p : m_instances) {
+        if (p.second->id() == id) {
+            return p.second;
+        }
+    }
+    return nullptr;
 }
 
-IVstInstancePtr VstInstancesRegister::instrumentPlugin(const muse::audio::TrackId trackId, const muse::audio::AudioResourceId& resourceId) const
+IVstInstancePtr VstInstancesRegister::instrumentPlugin(const muse::audio::AudioResourceId& resourceId,
+                                                       const muse::audio::TrackId trackId) const
 {
-    ONLY_MAIN_THREAD(threadSecurer);
```

Comment:
Why was it removed?

Reply:
here mutex is used everywhere for synchronization, i.e. access is safe from any thread.
Macro and securer are needed when the call is not protected, not safe, assumes a call only from a certain thread